x/sp/keeper/sp_status.go

```diff
@@ -0,0 +1,125 @@
+package keeper
+
+import (
+	"cosmossdk.io/errors"
+	"github.com/bnb-chain/greenfield/x/sp/types"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) UpdateToInMaintenance(ctx sdk.Context, sp *types.StorageProvider, requestDuration int64) error {
+	params := k.GetParams(ctx)
+	store := ctx.KVStore(k.storeKey)
+	newRecord := &types.MaintenanceRecord{
+		Height:          ctx.BlockHeight(),
+		RequestDuration: requestDuration,
+		RequestAt:       ctx.BlockTime().Unix(),
+	}
+	key := types.GetStorageProviderMaintenanceRecordsKey(sdk.MustAccAddressFromHex(sp.OperatorAddress))
+	bz := store.Get(key)
+	sp.Status = types.STATUS_IN_MAINTENANCE
+	if bz == nil {
+		if requestDuration > params.GetMaintenanceDurationQuota() {
```

Comment:
such a check should be done ASAP after you get the `param`.

Reply:
fixed